net.sourceforge.uhexen2.yml

```diff
@@ -0,0 +1,83 @@
+app-id: net.sourceforge.uhexen2
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
+sdk: org.freedesktop.Sdk
+command: /app/bin/uhexen2.sh
+
+add-extensions:
+  org.freedesktop.Platform.GL:
+    directory: lib/GL
+    version: '22.08'
+
+modules:
+  - shared-modules/glu/glu-9.json
+  - shared-modules/SDL/SDL-1.2.15.json
+  # Add this to compile with SDL2 support instead
+  # - shared-modules/SDL2/SDL-2-with-libdecor.json
+  - shared-modules/libmad/libmad.json
+
+  - name: uhexen2
+    sources:
+      - type: git
+        url: https://github.com/sezero/uhexen2
+        tag: uhexen2-1.5.9
+        commit: 4ef664bc41e3998b0d2a55ff1166dadf34c936be
+        x-checker-data:
+          type: git
+          tag-pattern: ^uhexen2-(.*)$
+    buildsystem: simple
+    build-commands:
+      # Uncomment this to apply the SDL2 patch
+      # - git apply patches/sdl2.patch
+      - cd engine/hexen2 && make glh2 && strip glhexen2
+      - install -D -m755 -t /app/bin/ ./engine/hexen2/glhexen2
+      - install -D -m755 -t /app/bin/data1/ ./gamecode/res/h2/*
+      - cd engine/hexen2 && make -C server && strip server/h2ded
+      - install -D -m755 -t /app/bin/ ./engine/hexen2/server/h2ded
+      - cd engine/hexenworld && make -C client glhw && strip client/glhwcl
+      - install -D -m755 -t /app/bin/ ./engine/hexenworld/client/glhwcl
+      - cd engine/hexenworld && make -C server && strip server/hwsv
+      - install -D -m755 -t /app/bin/ ./engine/hexenworld/server/hwsv
+      - cd hw_utils/hwmaster && make && strip hwmaster
+      - install -D -m755 -t /app/bin/ hw_utils/hwmaster/hwmaster
+
+  - name: launcher
+    sources:
+      - type: file
+        path: net.sourceforge.uhexen2.desktop
+      - type: file
+        path: net.sourceforge.uhexen2.appdata.xml
+      - type: file
+        path: net.sourceforge.uhexen2.png
+      - type: script
+        dest-filename: uhexen2.sh
+        commands:
+          - ls ~/.hexen2/data1/pak0.pak || zenity --error --text "Game data (the /data1/ directory)  must be added to:\n~/.var/app/net.sourceforge.uhexen2/.hexen2/data1/" --ok-label "Close" --width=400
+          - echo $1 $2
+          - /app/bin/glhexen2 "$@"
+    buildsystem: simple
+    build-commands:
+      - install -D -m755 -t /app/bin/ uhexen2.sh
+      - install -D -m644 -t /app/share/icons/hicolor/256x256/apps/ net.sourceforge.uhexen2.png
+      - install -D -m644 -t /app/share/applications/  net.sourceforge.uhexen2.desktop
+      - install -D -m644 -t /app/share/metainfo/ net.sourceforge.uhexen2.appdata.xml
+
+cleanup:
+  - /include
+  - /lib/*.a
+  - /lib/*.la
+  - /lib/pkgconfig
+
+finish-args:
+  # hardware 3D
+  - --device=dri
+  # X11 + XShm access
+  - --share=ipc
+  - --socket=x11
+  - --socket=fallback-x11
+  # Wayland access
+  - --socket=wayland
```

Comment:
does it support wayland? If yes then above line 77 should be kept ('fallback-x11'). Otherwise it should be remove along with this (line 79)